### ISSUE TITLE
Set-HVPool -PoolName '1531-28' -Disable does not work

### ISSUE THREAD

**Issue by TwoFaceSteph** [vmware/PowerCLI-Example-Scripts#177]
Hi,

I tried to disable a floating linked clone Pool named 1531-28 by this code:

Set-HVPool -PoolName '1531-28' -Disable

The Error Message is:

Updating the Pool:  1531-28
Exception calling "Desktop_Update" with "3" argument(s): "ExceptionType : VMware.Hv.InvalidArgument
ErrorMessage : Invalid value for member
ParameterName : automatedDesktopData.vmNamingSettings.patternNamingSettings.namingPattern"
At C:\Program Files\WindowsPowerShell\Modules\VMware.Hv.helper\VMware.HV.Helper.psm1:6171 char:8
+        $desktop_helper.Desktop_Update($services,$item,$updates)
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : VimException

I tried it by the pure API apporach but it looks similar:

Script (without the Connect-HV and other basic stuff):

function Get-MapEntry {
  param(
    [Parameter(Mandatory = $true)]
    $Key,
    [Parameter(Mandatory = $true)]
    $Value
  )
  $update = New-Object VMware.Hv.MapEntry
  $update.key = $key
  $update.value = $value
  return $update
}

$Pools = Get-HVQueryResult -EntityType DesktopSummaryView

foreach($desktop in $Pools)
{
     if($desktop.DesktopSummaryData.Source -eq "VIEW_COMPOSER" -and $desktop.DesktopSummaryData.Name -eq '1531-28')
    {
        $updates = Get-MapEntry -Key "desktopSettings.enabled" -Value $false
        $disablePool = $viewapi.Desktop.Desktop_Update($desktop.Id, $updates)
     }
}

Result:
Exception calling "Desktop_Update" with "2" argument(s): "ExceptionType : VMware.Hv.InvalidArgument
ErrorMessage : Invalid value for member
ParameterName : automatedDesktopData.vmNamingSettings.patternNamingSettings.namingPattern"
At C:\Scripte\powershell\VMware_View\View732\DisablePools.ps1:63 char:9
+         $disablePool = $viewapi.Desktop.Desktop_Update($desktop.Id, $ ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : VimException

Is it possible that the namingPattern has an issue with the name of the pool '1531-28'? Do I anything wrong?

My Pool Naming Pattern is: 
$test = $viewapi.Desktop.Desktop_Get($desktop.Id)
$test.AutomatedDesktopData.VmNamingSettings.PatternNamingSettings.NamingPattern

Results to (which is correct):
1531-28{n:fixed=3}

So maybe the "{}" are the problem?

**Comment by jackwmc4** [comment]
If you're naming pattern is 1531-28{n:fixed=3}, then this is incorrect syntax:
Set-HVPool -PoolName '1531-28' -Disable

It would need to be 
Set-HVPool -PoolName '1531-28001' -Disable

You'll want to enumerate your pools and disable them that way.

**Comment by TwoFaceSteph** [comment]
Hi jackwmc4,

I tried that out and here is the (expected) result:

Set-HVPool : No desktopsummarydata found with pool name: [1531-28001]
At C:\Scripte\powershell\VMware_View\View732\DisablePools.ps1:48 char:1
+ Set-HVPool -Poolname '1531-28001' -Disable
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Set-HVPool
This is what I expected, because the name of the pool is 1531-28 and not 1531-28001. 1531-28001 is the first VM within that pool, because of the naming pattern for this pool.

**Comment by TwoFaceSteph** [comment]
This what I get calling Get-HVPoolSummary:

Name         DisplayName      Type         Source             User Assignment  Entitled Enabled   Sessions
----         -----------      ----         ------             ---------------  -------- -------   --------
1511-11      HCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      DEDICATED        1        True             0
1511-20      HCLT x64 Pool 20 AUTOMATED    VIEW_COMPOSER      DEDICATED        1        True            19
1511-21      HCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      DEDICATED        1        True             0
1511-22      HCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      DEDICATED        1        True             0
1511-23      HCLT x64 Pool 23 AUTOMATED    VIEW_COMPOSER      DEDICATED        2        True            24
1511-24      HCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      DEDICATED        4        True             1
1512-11      HCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      DEDICATED        2        True             0
1521-20      ACLT x64 (Win... AUTOMATED    VIRTUAL_CENTER     DEDICATED        4        True             0
1521-22      ACLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True             0
1521-23      ACLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True            16
1521-24      ACLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True             7
1521-25      ACLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True             1
1531-20      SCLT x64 Pool 20 AUTOMATED    VIEW_COMPOSER      FLOATING         1        True            26
1531-21      SCLT x64 Pool 21 AUTOMATED    VIEW_COMPOSER      FLOATING         1        True            26
1531-22      SCLT x64 Pool 22 AUTOMATED    VIEW_COMPOSER      FLOATING         1        True            32
1531-23      SCLT x64 Pool 23 AUTOMATED    VIEW_COMPOSER      FLOATING         1        True            34
1531-24      SCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      FLOATING         2        True             6
1531-25      SCLT x64 Pool 25 AUTOMATED    VIEW_COMPOSER      FLOATING         1        True            25
1531-26      SCLT x64 Pool 26 AUTOMATED    VIEW_COMPOSER      FLOATING         1        True            23
1531-28      SCLT x64 Pool... AUTOMATED    VIEW_COMPOSER      FLOATING         3        True             0
1533-21      Thinapp Clien... AUTOMATED    VIEW_COMPOSER      FLOATING         1        True             0
1541-24      ECLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        2        True            15
1541-25      ECLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        3        True            17
1541-26      ECLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True             1
1551-20      TCLT x64 Pool... MANUAL       VIRTUAL_CENTER     DEDICATED        2        True             0
1561-23      DCLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True            18
1561-24      DCLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True            15
1571-21      CCLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True             9
1571-22      CCLT x64 Pool... AUTOMATED    VIRTUAL_CENTER     DEDICATED        1        True            10

**Comment by jackwmc4** [comment]
Gotcha, then likely there might be a bug with the parsing of the names but I'll have to look into it later this week when I have time. You might look through the code and see if the dash separator is causing an issue.

**Comment by TwoFaceSteph** [comment]
I don't think that the code "Set-HVPool" is the problem. Set-HVPool is calling the api method: Desktop.Desktop_Update($services,arg,arg) right? So tried that directly with the api, but using the method Desktop_update($desktop.Id, [key,value]) leads to the same error message, so I think the api itself has an issue with that kind of pool name. Don't know how to debug the api, just able to debug ps code, sorry :-)

**Comment by freythman** [comment]
`1531-28{n:fixed=3}` is a valid naming convention. I, too, suspect that this is related to the API misinterpreting the dash in the pool name. When I am able to get my lab going again I'll test that theory.

**Comment by jackwmc4** [comment]
> so I think the api itself has an issue with that kind of pool name. Don't know how to debug the api, just able to debug ps code, sorry :-)

You could try modifying the naming pattern and inserting escape characters so \{ instead of { to see if that makes a difference. That's where I would start. Once we can track where the issue is I can open an internal bug on the API.

**Comment by jackwmc4** [comment]
I quickly tried to recreate this and this is infact a bad name. Is this by chance an environment that was upgraded from View 5.x?

![2018-03-20_17-35-59](https://user-images.githubusercontent.com/1228172/37688359-4418f87a-2c65-11e8-9c86-b8c5a60ff014.png)

**Comment by TwoFaceSteph** [comment]
We are coming from View 6.2.4 upgrades directly to 7.3.2. 
The Weird thing is that disabling or modifying the pool within the Admin UI works without any trouble, by the way we are using linked clones at this time beacuse Instant Clones need hardware version 11 on ESXi 6.5 and we are still on ESXi 5.5 and therefore hardware version 10. 
So the question is: Where is the gap between the Admin UI and the API?

**Comment by TwoFaceSteph** [comment]
Forgot to mention that we are chosing the pool names by a name convention:

First two digits:  Clients (here 15)
Second two digits: Client Type (31 = Default Office Client)
Fith digit: Separator for Pools: -
Sixth and seventh digits are counter for the pools, so for all admins the Poolname 1531-28 points out that it is a client pool for default office clients and it is the 28th pool.
So it is not an option to change all our pool names.

**Comment by jackwmc4** [comment]
> The Weird thing is that disabling or modifying the pool within the Admin UI works without any trouble

That's because it isn't revalidating it. Did you create the pools programatically? This isn't a new issue, and the Admin UI and API are consistent. The Admin UI throws the same error because it uses the same API. It has been an issue for some time now that computer names require at least one letter, but I can't remember why. If you add a letter to your naming convention your issue will go away. So this issue isn't an API or PowerCLI issue, and should be closed. You'll want to open a support request with VMware and track the issue that way.

**Comment by TwoFaceSteph** [comment]
I created all those pools with the Admin UI and there was no error during creation of the pool. I tested this again by creating a new linked clone floating pool with name 1531-30 and the naming pattern 1531-30{n:fixed=3} and it worked like a charm. The error only occurs using PowerCLI and by the way with the old powershell cmdlets that came with version 6.2.4 this worked well, this issue comes up using the new API integrated within the new PowerCLI and Horizon 7.3.2 (old cmdlets do sadly not working anymore), so if both the UI and the API works the same, why does it work within the UI but not in PowerCLI?
Nevertheless I will open a support request.
